File: Python/Esercitazione10/materiale.py
class Materiale:
	def __init__(self, nome : str, quantita : int, prezzo_unitario : float):
		self._nome = nome
		self._quantita = quantita
		self._prezzo_unitario = prezzo_unitario


	def get_nome(self) -> str:
		return self._nome

	def get_quantita(self) -> int:
		return self._quantita

	def get_prezzo_unitario(self) -> float:
		return self._prezzo_unitario

	def set_nome(self, nome : str) -> None:
		self._nome = nome

	def set_quantita(self, quantita : int) -> None:
		self._quantita = quantita

	def set_prezzo_unitario(self, prezzo_unitario : float) -> None:
		self._prezzo_unitario = prezzo_unitario


	def __repr__(self):
		return f"Materiale[nome='{self._nome}'; quantità={self._quantita}; prezzo unitario={self._prezzo_unitario}]"

File: Java/Esercitazione10/Materiale.java
package Esercitazione10;

import java.io.Serial;
import java.io.Serializable;

public class Materiale implements Serializable {
    @Serial
    private static final long serialVersionUID = 1L;
    private String nome;
    private int quantita;
    private double prezzoUnitario;

    

    public Materiale(String nome, int quantita, double prezzoUnitario) {
        this.nome = nome;
        this.quantita = quantita;
        this.prezzoUnitario = prezzoUnitario;
    }



    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public int getQuantita() {
        return quantita;
    }

    public void setQuantita(int quantita) {
        this.quantita = quantita;
    }

    public double getPrezzoUnitario() {
        return prezzoUnitario;
    }

    public void setPrezzoUnitario(double prezzoUnitario) {
        this.prezzoUnitario = prezzoUnitario;
    }



    @Override
    public String toString() {
        return "Materiale{" +
                "nome='" + nome + '\'' +
                ", quantita=" + quantita +
                ", prezzoUnitario=" + prezzoUnitario +
                '}';
    }
}
